byterun/io.c

```diff
@@ -118,25 +154,45 @@ CAMLexport void caml_close_channel(struct channel *channel)
   unlink_channel(channel);
   caml_stat_free(channel->name);
   caml_stat_free(channel);
+  return;
+}
+
+static io_result caml_lseek_end(int fd, file_offset* end)
+{
+  caml_enter_blocking_section();
+  *end = lseek(fd, 0, SEEK_END);
+  caml_leave_blocking_section_nosig();
+  if (*end == -1) return errno;
+  return 0;
+}
+
+static io_result caml_lseek_set(int fd, file_offset offset)
+{
+  file_offset off;
+  off = lseek(fd, offset, SEEK_SET);
```

Comment:
You need a blocking section around this call to `lseek` because it is a blocking system call, and indeed the original code has both calls to `lseek` inside a blocking section.